### ISSUE TITLE
2024 DStability changes

### ISSUE THREAD

**Issue by breinbaas** [Deltares/GEOLib#170]
Creating a DStability model and then trying to calculate it using 2024.01 results in the following error;

```
Output file generated but parsing of ... 34-1_0.45-0.81_0.stix failed.

pydantic.error_wrappers.ValidationError: 2 validation errors for UpliftVanParticleSwarmResult
ResultThreshold
  extra fields not permitted (type=value_error.extra)
SlipPlaneResults
  extra fields not permitted (type=value_error.extra)
```

This is due to missing fields in the geolib code.

**Comment by breinbaas** [comment]
fixed for upliftvanparticleswarm -> needed to set 

```
ResultThreshold: float = 0.0
```

instead of 

```
ResultThreshold: Optional[float] = None
```

because the latter leads to models that cannot be opened in 2024.01

**Comment by breinbaas** [comment]
fixed bishop brute force / spencer genetic algorithm where field were added as well

**Comment by breinbaas** [comment]
There seems to be a bugfix already; https://github.com/Deltares/GEOLib/pull/163 

Not sure which code is better!

**Comment by wfaustmann** [comment]
Fixed in GEOLib v2.3.0
